qiskit/namespace.py

```diff
@@ -43,16 +45,23 @@ def load_module(self, fullname):
         fullname = _new_namespace(fullname, self.old_namespace, self.new_package)
         module = importlib.import_module(fullname)
         sys.modules[old_name] = module
+        if self.deprecate:
+            warnings.warn(
+                f"Importing {old_name} is deprecated and will be removed in "
+                f"a future release. Instead you should import {fullname}.",
+                DeprecationWarning,
+            )
```

Comment:
Not sure if we want to bother trying to unroll the stack enough to point to the right `import` statement?  I'm guessing it'd be like stacklevel=12 or something insane, by the time it's worked its way through all the `importlib` machinery and the containing class.  Probably not worth it, if so.

Reply:
Yeah, I didn't even want to attempt to figure out how deep in the call stack this is from an `import` call. I also wasn't convinced it'd always be a stable stack size as I think this loader might get called in different ways depending on the circumstances (but I'm not exactly sure).